packages/router/src/utils/collection.ts

```diff
@@ -20,12 +20,14 @@ export function shallowEqualArrays(a: any[], b: any[]): boolean {
 }
 
 export function shallowEqual(a: Params, b: Params): boolean {
-  // Casting Object.keys return values to include `undefined` as there are some cases
-  // in IE 11 where this can happen. Cannot provide a test because the behavior only
-  // exists in certain circumstances in IE 11, therefore doing this cast ensures the
-  // logic is correct for when this edge case is hit.
-  const k1 = Object.keys(a) as string[] | undefined;
-  const k2 = Object.keys(b) as string[] | undefined;
+  // Previously cast Object.keys return values to include `undefined` as there
+  // are some cases in IE 11 where this can happen. Cannot provide a test
+  // because the behavior only exists in certain circumstances in IE 11,
+  // therefore doing this cast ensures the logic is correct for when this edge
+  // case is hit. Adding a check for a and b being defined because in Edge
+  // sometimes Object.keys throws an error.
```

Comment:
```suggestion
  // While `undefined` should never be possible, it would sometimes be the case in IE 11 
  // and pre-chromium Edge. The check below accounts for this edge case.
```
This comment is getting pretty long and confusing. Maybe we can just simplify it like this?